numpy/_core/multiarray.pyi

```diff
@@ -199,6 +218,72 @@ def array(
     like: None | _SupportsArrayFunc = ...,
 ) -> NDArray[_SCT]: ...
 @overload
+def array(
+    object: _ArrayLikeBool_co,
+    dtype: None = ...,
+    *,
+    copy: None | bool | _CopyMode = ...,
+    order: _OrderKACF = ...,
+    subok: bool = ...,
+    ndmin: int = ...,
+    like: None | _SupportsArrayFunc = ...,
+) -> NDArray[np.bool]: ...
+@overload
+def array(
+    object: _ArrayLikeUInt_co,
+    dtype: None = ...,
+    *,
+    copy: None | bool | _CopyMode = ...,
+    order: _OrderKACF = ...,
+    subok: bool = ...,
+    ndmin: int = ...,
+    like: None | _SupportsArrayFunc = ...,
+) -> NDArray[unsignedinteger[Any]]: ...
+@overload
+def array(
+    object: _ArrayLikeInt_co,
+    dtype: None = ...,
+    *,
+    copy: None | bool | _CopyMode = ...,
+    order: _OrderKACF = ...,
+    subok: bool = ...,
+    ndmin: int = ...,
+    like: None | _SupportsArrayFunc = ...,
+) -> NDArray[signedinteger[Any]]: ...
+@overload
+def array(
+    object: _ArrayLikeFloat_co,
+    dtype: None = ...,
+    *,
+    copy: None | bool | _CopyMode = ...,
+    order: _OrderKACF = ...,
+    subok: bool = ...,
+    ndmin: int = ...,
+    like: None | _SupportsArrayFunc = ...,
+) -> NDArray[floating[Any]]: ...
+@overload
+def array(
+    object: _ArrayLikeComplex_co,
+    dtype: None = ...,
+    *,
+    copy: None | bool | _CopyMode = ...,
+    order: _OrderKACF = ...,
+    subok: bool = ...,
+    ndmin: int = ...,
+    like: None | _SupportsArrayFunc = ...,
+) -> NDArray[complexfloating[Any, Any]]: ...
+@overload
+def array(
+    object: _ArrayLikeTD64_co,
+    dtype: None = ...,
+    *,
+    copy: None | bool | _CopyMode = ...,
+    order: _OrderKACF = ...,
+    subok: bool = ...,
+    ndmin: int = ...,
+    like: None | _SupportsArrayFunc = ...,
+) -> NDArray[timedelta64]: ...
```

Comment:
These `_ArrayLike{}_co` aliases should be replaced with the `T | _NestedSequence[T]` part from [their definition](https://github.com/guan404ming/numpy/blob/55c10be2ebb451b80924016a3628866c857827b8/numpy/_typing/_array_like.py#L96-L128), because they also include `_ArrayLike`, which you already covered.

So e.g. `_ArrayLikeFloat_co` will become `float | _NestedSequence[float]` (the `int` and `bool` are implied by `float`, because the type param of `_NestedSequence` is covariant). 
And because `float` could also come as a union with `int` or `bool`, the return type should be changed to `NDArray[np.bool | np.int_ | np.float64]` (note the narrowed `floating` and `integer`), so that it's always compatible with the (previous) overload for `int | _NestedSequence[int]`.